packages/apps/composer-app/src/main.tsx

```diff
@@ -15,7 +15,7 @@ import { initializeAppTelemetry } from '@dxos/react-appkit/telemetry';
 
 import { createRouter, namespace } from './router';
 
-void initializeAppTelemetry({ namespace, config: new Config(Defaults()) });
+await initializeAppTelemetry({ namespace, config: new Config(Defaults()) });
```

Comment:
Is it safe to do @wittjosiah?

Reply:
It will be less painless for now to avoid top-level await and if you need app telemetry to finish initializing before client starts then do something like this:

```
const main = async () => {
    await initializeAppTelemetry({ namespace, config: new Config(Defaults()) });

    createRoot(document.getElementById('root')!).render(
    <StrictMode>
        <App />
    </StrictMode>
    );
};

void main();
```

Keep in mind though, that in general the telemetry should never get in the way of the app experience (which I think awaiting it in order to render the app potentially could). So if this is a temporary thing in order to ensure we're capturing certain things in telemetry to improve stability, that's great, but if we're thinking the telemetry init needs to be this way forever we way want to think through how that is functioning.